### PR TITLE
Add instrumentation provider

### PR DESCRIPTION
**Pull request by jlegoff**
This PR sets the `instrumentation.provider` attribute to `newrelic-opentelemetry` so as to switch on the new UI for the data flowing through these components.

### CODE REVIEW

File: connector/apmconnector/transaction.go [hunk 1]
```diff
@@ -355,9 +352,8 @@ func GetTransactionMetricName(span ptrace.Span) (string, TransactionType) {
 	if rpcService, rpcServicePresent := span.Attributes().Get("rpc.service"); rpcServicePresent {
 		if rpcMethod, rpcMethodPresent := span.Attributes().Get("rpc.method"); rpcMethodPresent {
 			return fmt.Sprintf("WebTransaction/rpc/%s/%s", rpcService.AsString(), rpcMethod.AsString()), WebTransactionType
-		} else {
-			return fmt.Sprintf("WebTransaction/rpc/%s", rpcService.AsString()), WebTransactionType
 		}
+		return fmt.Sprintf("WebTransaction/rpc/%s", rpcService.AsString()), WebTransactionType
```
Comment (sdaubin) [inline connector/apmconnector/transaction.go:356]: 😆

File: processor/apmprocessor/config.go [hunk 2]
```diff
@@ -8,6 +8,8 @@ import (
 )
 
 type Config struct {
+	// If set to true, will set the `instrumentation.provider` attribute to `newrelic-opentelemetry`
```
Comment (sdaubin) [inline processor/apmprocessor/config.go:11]: We should add a comment that the default is true

**Review by sdaubin — DISMISSED** [review]

File: connector/apmconnector/transaction.go (hunk 1 above)
Comment (jlegoff) [reply connector/apmconnector/transaction.go:356]: I didn't do this on a whim :) running `make all` breaks if I don't remove this `else` condition

File: processor/apmprocessor/config.go (hunk 2 above)
Comment (jlegoff) [reply processor/apmprocessor/config.go]: 👍 done

**Review by sdaubin — APPROVED** [review]
Works now.  Thanks!

**Pull request merged**